crates/router/src/db/api_keys.rs

```diff
@@ -22,11 +22,16 @@ pub trait ApiKeyInterface {
 
     async fn revoke_api_key(&self, key_id: &str) -> CustomResult<bool, errors::StorageError>;
 
-    async fn find_api_key_optional(
+    async fn find_api_key_by_key_id_optional(
```

Comment:
is this function is required?

Reply:
Yes, retrieval, updation and deletion of API keys happen by key ID. Authentication involves fetching API key by API key hash.